models/_sector/depin/geodnet/geodnet_revenue.sql

```diff
@@ -0,0 +1,127 @@
+{{ config(
+    schema = 'geodnet'
+    , alias = 'revenue'
+    , materialized = 'incremental'
+    , file_format = 'delta'
+    , incremental_strategy = 'merge'
+    , unique_key = ['date', 'chain', 'name']
+    )
+}}
+
+WITH
+  hour_series AS (
+    SELECT
+      time AS hour,
+      DATE_TRUNC('day', time) AS day
+    FROM
+      UNNEST (
+        sequence(
+          CAST('2023-04-20 00:00' AS timestamp),
+          CAST(
+            DATE_TRUNC('hour', CURRENT_TIMESTAMP) AS timestamp
+          ),
+          INTERVAL '1' hour
+        )
+      ) AS t (time)
+      {% if is_incremental() %}
+      WHERE
+          time >= (
+            SELECT
+              CAST(COALESCE(MAX(date), '2023-04-27 00:00') AS timestamp) - INTERVAL '7' DAY
+            FROM
+              {{ this }}
+          )
+      {% endif %}
+  ),
+  burn_events AS (
+    SELECT
+      DATE_TRUNC('hour', a.evt_block_time) AS evt_hour,
+      DATE_TRUNC('day', a.evt_block_time) AS evt_day,
+      CAST(value AS double) / pow(10, b.decimals) AS tokens_burned
+    FROM {{ source('erc20_polygon', 'evt_transfer') }} a
+      JOIN {{ source('tokens', 'erc20') }} b ON a.contract_address = b.contract_address
+    WHERE
+      a.contract_address = 0xAC0F66379A6d7801D7726d5a943356A172549Adb
+      AND a.to = 0x000000000000000000000000000000000000dead
+      {% if is_incremental() %}
+      AND a.evt_block_time >= (
+        SELECT
+          CAST(COALESCE(MAX(date), '2023-04-27 00:00') AS timestamp) - interval '7' day
+        FROM
+          {{ this }}
+      )
+      {% endif %}
+  ),
+  raw_prices AS (
+    SELECT
+      AVG(median_price) AS price,
+      DATE_TRUNC('hour', hour) AS hour,
+      DATE_TRUNC('day', hour) AS day
+    FROM {{ ref('dex_prices') }}
+    WHERE
+      blockchain = 'polygon'
+      AND contract_address = 0xac0f66379a6d7801d7726d5a943356a172549adb
+      {% if is_incremental() %}
+      AND hour >= (
+        SELECT
+          CAST(COALESCE(MAX(date), '2023-04-27 00:00') AS timestamp) - interval '7' day
+        FROM
+          {{ this }}
```

Comment:
we have noticed performance issues on trino when attempting to filter incrementally on a subquery that reads from target. this is why we use the standard incremental macro across spellbook. i'd advise we update to that standard